libPhoneNumberGeocoding/Metadata/NBGeocoderMetadataHelper.h

```diff
@@ -0,0 +1,50 @@
+//
+//  NBGeocoderMetadataHelper.h
+//  libPhoneNumberiOS
+//
+//  Created by Rastaar Haghi on 6/12/20.
+//  Copyright © 2020 Google LLC. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import <sqlite3.h>
+
+@class NBPhoneNumber;
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface NBGeocoderMetadataHelper : NSObject
+
+/**
+ * Initializer method that creates a SQLite3 connection to the correct database (based on
+ * the language passed in as a parameter), which is stored in the resource bundle of this project.
+ * This method opens the database for viewing if possible and prepares a query statement with
+ * the country code parameter
+ *
+ * @param countryCode a valid country code for which we want to set up a prepared statement to
+ *  access a SQLite table with that country code.
+ * @param languageCode  the language code for which the description should be written
+ * @return a NBGeocoderMetadataHelper instance variable with an initial country code and language
+ *     code set to the parameters, countryCode and  languageCode, or nil if the NSObject couldn't be
+ *     initialized or there was no database URL found for the provided language code.
+ */
+- (instancetype)initWithCountryCode:(NSNumber *)countryCode withLanguage:(NSString *)languageCode;
+
+/**
+ * Returns a text description for the given phone number. The description will be based on the
+ * country code and national number attributes of the NBPhoneNumber parameter. If no result was
+ * found from querying the geocoding databases, this method will return nil.
+ *
+ * This method assumes the validity of the NBPhoneNumber object passed in that it contains a
```

Comment:
I feel like for this one, it's fine to mention SQLite because it's a header file to the .m file, rather than a protocol to an implementation, though it kinda counts as an API doc, it's still really part of the implementation. wdyt?

Reply:
lol objective-c. I forgot headers aren't protocols. 🤦